src/main/java/randoop/generation/Bloodhound.java

```diff
@@ -0,0 +1,206 @@
+
+      // Use default weight if no coverage details are available.
+      // This is the case for classes like java.lang.Object (not explicitly under test).
+      if (covDet != null) {
+
+        // The number of successful invocations of this method. Corresponds to succ(m).
+        Integer numSuccessfulInvocation = methodSuccCalls.get(operation);
+        if (numSuccessfulInvocation == null) {
+          numSuccessfulInvocation = 0;
+        }
+
+        // If this method has no uncovered branches, or it's been invoked the current maximum number
+        // of times, use the default weight and skip this step.
+        if (covDet.uncoveredBranches != 0 && numSuccessfulInvocation != maxSuccessfulCalls) {
+          // Uncovered branch ratio of this method. Corresponds to uncovRatio(m) in the GRT paper.
+          double uncoveredRatio = 0.5;
+          if (covDet.numBranches != 0) {
+            uncoveredRatio = (double) covDet.uncoveredBranches / covDet.numBranches;
+          }
+
+          // Call ratio of this method. Corresponds to succ(m) / maxSucc(M) in the GRT paper.
+          double callRatio = 0.5;
+          if (maxSuccessfulCalls != 0) {
+            callRatio = numSuccessfulInvocation.doubleValue() / maxSuccessfulCalls;
+          }
+
+          // Corresponds to w(m, 0) in the GRT paper.
+          weight = alpha * uncoveredRatio + (1 - alpha) * (1 - callRatio);
+
+          // Corresponds to the k variable in the GRT paper.
+          Integer numSelectionsOfMethod = methodSelections.get(operation);
+          if (numSelectionsOfMethod != null) {
+            // Corresponds to the case where k >= 1 in the GRT paper.
+            double val1 =
+                (-3.0 / Math.log(1 - p))
+                    * (Math.pow(p, numSelectionsOfMethod) / numSelectionsOfMethod);
+            double val2 = 1.0 / Math.log(numOperations + 3);
+            weight *= Math.max(val1, val2);
+          }
+        }
+      }
+
+      // Assign the weight to the operation in the methods' weight map.
+      methodWeights.put(operation, weight);
+    }
+  }
+
+  /**
+   * First, update the weights of all methods under test. Retrieve the next method for constructing
+   * a new sequence while also considering each method's weights. Update the number of times the
+   * method has been selected.
+   *
+   * @return the chosen {@code TypedOperation} for the new sequence
+   */
+  @Override
+  public TypedOperation selectNextOperation() {
+    // Collect branch coverage and recompute weights for methods under test.
+    updateBranchCoverageAtInterval();
+    updateWeightsForOperations();
+
+    TypedOperation operation = Randomness.randomMemberWeighted(operationSimpleList, methodWeights);
+
+    // Update the number of times this method was selected for a new sequence.
+    incrementInMap(methodSelections, operation);
+
+    // Update the number of times this method was successfully invoked.
```

Comment:
This line speaks of "successfully invoked", but just above it's "selected".  These must be the same concept, because one is incremented whenever the other is.  Why are two different terms used for it?